### PR TITLE
Improve UX of layer style categories with html 

### PR DESCRIPTION
**Pull request by signedav**
What means we provide displayRole in html format and have a delegate for the display text.

## Description

There is a mess in the categories (see https://github.com/qgis/QGIS/issues/58432 and https://github.com/qgis/QGIS/issues/45146) 

To reorder them would break backwards compatibility (what doesn't mean that we should not do it). But meanwhile this UX only improvement should help in the meantime.

image

- The grey texts are ***more descriptive*** and always there (because tooltips are nice, but if you don't know about them you won't look for them)
- There are new icons for ***layer configuration*** (without being mistaken with the *source*, ***fields*** (contains a small forms in there, to indicate it's a little bit more than just the fields) and ***notes*** (was missing before)
- Model and Delegate are now available in the API

- [x] Improve design (grey)
- [x] word wrap
- [x] Improve texts with info from https://github.com/qgis/QGIS/issues/45146#issuecomment-2345813144
- [x] maybe better icons ( an notes icon)
- [x] expose in the API (to ensure plugins using the same)

### CODE REVIEW

**Comment by kannes** [comment]
Nice! I think it could look a bit less busy if the longer, informative lines used a grey or thinner or smaller font.

**Comment by signedav** [comment]
> Nice! I think it could look a bit less busy if the longer, informative lines used a grey or thinner or smaller font.

Thanks. I tried italic and it seemed noisy. Maybe grey would be worth a try...

**Comment by nirvn** [comment]
@signedav , good move, that'll help guide people through this otherwise relatively obscure set of categories. +1 to maybe have the text grayed out and/or slightly smaller (relative to main font app size).

**Comment by signedav** [comment]
Opinions on that @nirvn @kannes ?

|grey|small|grey-and-small|
|---|---|---|
|Screenshot from 2024-09-14 23-08-37|Screenshot from 2024-09-14 23-09-49|Screenshot from 2024-09-14 23-05-02|

I like the look of the smaller font, but I would rather "not read" it than the grey one.
Anyway I'm -1 to have small *and* grey.

**Review by pathmapper — COMMENTED** [review]
Thank you, really helpful! 

Maybe add also an icon for "Notes" to have the text aligned with the other categories? (a sticky note symbol would perhaps fit)

I'm +1 for the `grey` variant, but would also be fine with the others as this is in general an improvement.

**Comment by kannes** [comment]
+1 for grey

File: src/gui/qgsmaplayerstylecategoriesmodel.cpp [hunk 1]
```diff
@@ -110,219 +110,350 @@ QVariant QgsMapLayerStyleCategoriesModel::data( const QModelIndex &index, int ro
...
         case Qt::DecorationRole:
           return QgsApplication::getThemeIcon( QStringLiteral( "/propertyicons/relations.svg" ) );
       }
       break;
-
+    }
     case QgsMapLayer::StyleCategory::Temporal:
+    {
+      QString name = tr( "Temporal Properties" );
+      QString description = tr( "Temporal Properties" );
```
Comment (nyalldawson) [inline src/gui/qgsmaplayerstylecategoriesmodel.cpp:370]: There's a mix of style here -- should descriptions always be title case? It is here, but not for eg relations. I'd opt for sentence case consistently instead.

Comment (signedav) [reply src/gui/qgsmaplayerstylecategoriesmodel.cpp]: thanks. I'll do that.

**Comment by fiddlersfan** [comment]
Nice work, thanks! Just a small visual inconstancy in night mapping at the note icon: 

I would suggest coloring the paper white here, similar to the clock icon. Unfortunately, we can then no longer use the mIndicatorNotes icon, which is used as a monochrome icon in the layer overview.

**Comment by signedav** [comment]
Considered that @fiddlersfan

**Comment by signedav** [comment]
Can we merge that? All inputs have been considered.

**Review by nirvn — APPROVED** [review]
Let's merge this now, I'll give the notes icon a try in the coming weeks.

**Pull request merged**